src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -655,7 +654,7 @@ private void GetEnumOperations(BinaryOperatorKind kind, BoundExpression left, Bo
                 GetPointerArithmeticOperators(kind, rightType, results);
             }
 
-            if ((object)leftType != null || (object)rightType != null)
+            if ((object)leftType != null || (object)rightType != null || left.Type is FunctionPointerTypeSymbol || right.Type is FunctionPointerTypeSymbol)
             {
                 // The pointer comparison operators are all "void* OP void*".
                 GetPointerComparisonOperators(kind, results);
```

Comment:
>GetPointerComparisonOperators [](start = 16, length = 29)

Do we support `<` and `>` for function pointers but not pointer arithmetic?

Reply:
Unfortunately, yes. I was considering trying to not support `>` and `<`, but they're defined for `void*` so I don't have an option there. But the arithmetic operators are defined for specific pointer types except `void*`, so we can say "these are not defined for function pointer types".

---
In reply to: [410505291](https://github.com/dotnet/roslyn/pull/43355#discussion_r410505291) [](ancestors = 410505291)